src/vmm/src/persist.rs

```diff
@@ -330,6 +350,14 @@ fn guest_memory_from_file(
     GuestMemoryMmap::restore(&mem_file, mem_state, track_dirty_pages).map_err(DeserializeMemory)
 }
 
+fn validate_devices_number(device_number: usize) -> std::result::Result<(), CreateSnapshotError> {
+    use self::CreateSnapshotError::TooManyDevices;
+    if device_number > FC_V0_23_MAX_DEVICES as usize {
+        return Err(TooManyDevices(device_number));
+    }
+    Ok(())
+}
```

Comment:
nit: this is function is specific to v0.23-only, it've better been hardcoded in the version `match` or at least it should have `v023` in its name.